mage_integrations/mage_integrations/tests/sources/test_base.py

```diff
@@ -0,0 +1,473 @@
+from mage_integrations.sources.catalog import Catalog, CatalogEntry
+from mage_integrations.sources.base import Source
+from mage_integrations.sources.intercom import Intercom
+from mage_integrations.sources.postgresql import PostgreSQL
+from mage_integrations.sources.stripe import Stripe
+from singer.schema import Schema
+from unittest.mock import patch
+import os
+import unittest
+import json
+import sys
+
+
+def build_sample_streams_catalog_entries():
+    return [
+        CatalogEntry(
+            stream='demo_table',
+            tap_stream_id='demo_table',
+        ),
+        CatalogEntry(
+            stream='demo_users',
+            tap_stream_id='demo_users',
+        ),
+    ]
+
+def build_sample_streams_list():
+    ABSOLUTE_PATH = os.path.abspath(os.path.dirname(__file__))
```

Comment:
the constant variable is usually not defined within method. you can define it below the import statements.